airflow/providers/google/cloud/hooks/cloud_run.py

```diff
@@ -0,0 +1,233 @@
+            raise job_exec_exception
+        execution = cast(Dict[str, Union[str, dict, int]], self._execution)
+        if "status" not in execution.keys():
+            raise job_exec_exception
+        status = cast(Dict[str, Union[int, str, List[dict]]], execution["status"])
+        if "conditions" not in status.keys():
+            raise job_exec_exception
+        conditions = cast(List[Dict[str, str]], status["conditions"])
+        steps = []
+        for c in conditions:
+            step = c["type"]
+            verdict = c["status"]
+            if step not in CloudRunJobSteps.ALL_STEPS:
+                raise Exception(
+                    f"Unknown state {step} found for Google Cloud Run job execution {self._exec_id}. "
+                    f"See details at {logs_uri}."
+                )
+            if verdict == "False":
+                # This verdict means that a step failed,
+                # therefore the execution has failed
+                raise Exception(
+                    f"Cloud Run Job execution {self._exec_id} has failed. See details at {logs_uri}"
+                )
+            if verdict == "Unknown":
+                # This verdict means that the step is not completed yet,
+                # therefore the execution cannot be finished yet
+                return False
+            if verdict == "True":
+                steps.append(step)
+        return CloudRunJobSteps.EXEC_STEP_COMPLETED in steps
+
+    def wait_for_done(self) -> None:
+        """Helper method to wait for result of job execution."""
+        # Wait a few seconds for the job execution status to be available
+        # otherwise it fails
+        time.sleep(self._job_exec_cold_start)
+        if self._wait_until_finished:
+            self.log.info("Starting to poll status for Google Cloud Run job execution %s", self._exec_id)
+            self._execution = self._fetch_execution_by_id()
+            self.log.info(json.dumps(self._execution))
+            while not self._check_execution_state():
+                self.log.info("Waiting for execution completion. Sleeping %d seconds", self._poll_sleep)
+                time.sleep(self._poll_sleep)
+                self._execution = self._fetch_execution_by_id()
+        else:
+            # Check execution state only once to ensure the job has started
+            # Otherwise, _check_execution_state method would have raised an error
+            self._check_execution_state()
+            return
+
+
+class CloudRunJobHook(GoogleBaseHook):
+    """
+    Hook for Google Cloud Run.
+
+    All the methods in the hook where project_id is used must be called with
+    keyword arguments rather than positional.
+    :param gcp_conn_id: The Google connection ID to use to make Cloud Run API calls
+    :param wait_until_finished: If True, wait for the end of pipeline execution before exiting.
+        If False,it only submits job and check once is job not in terminal state.
```

Comment:
```suggestion
        If False, it only submits job and check once is job not in terminal state.
```

And also found in another place